contracts/contracts/structs/CrossNet.sol

```diff
@@ -29,6 +29,31 @@ struct BottomUpCheckpoint {
     uint64 nextConfigurationNumber;
     /// @dev Batch of messages to execute.
     IpcEnvelope[] msgs;
+    /// @dev A commitment to the summary of our chain activity since the previous checkpoint and this one.
+    bytes32 summary;
+    /// @dev Summaries relayed upwards from descendants of this subnet.
+    /// NOTE: Not merkelized to keep it simple, but we will merkelize later to scale better.
+    RelayedSummary[] relayedSummaries;
+}
+
+struct ActivitySummary {
+    /// @dev The block range the activity summary spans; these are the local heights of the start and the end, inclusive.
+    uint256[2] blockRange;
+    /// @dev The validators whose activity we're reporting about.
+    address[] validators;
```

Comment:
Just curious why 2 arrays instead of a mapping?

Reply:
I understand the arrays if when distributing the rewards we have to process them all but I agree with @cryptoAtwill, validators could claim their own rewards, so a mapping would make sense.

Reply:
This is a DTO, so mapping cannot, but I have updated the data struct slightly for storage